lib/vfscore/mount.c

```diff
@@ -171,8 +171,16 @@ UK_SYSCALL_R_DEFINE(int, mount, const char*, dev, const char*, dir,
 	mp->m_flags = flags;
 	mp->m_dev = device;
 	mp->m_data = NULL;
-	strlcpy(mp->m_path, dir, sizeof(mp->m_path));
-	strlcpy(mp->m_special, dev, sizeof(mp->m_special));
+	mp->m_path = strndup(dir, PATH_MAX - 1);
+	if (!mp->m_path) {
+		error = ENOMEM;
+		goto err2;
```

Comment:
On errors `mp->m_path` and `mp->m_special` are empty, so you should go to free'ing `mp`, instead.
```c
goto err_free_mp;
```